src/data/finset/lattice.lean

```diff
@@ -279,12 +279,20 @@ theorem min'_le (x) (H2 : x ∈ s) : s.min' H ≤ x := min_le_of_mem H2 $ option
 
 theorem le_min' (x) (H2 : ∀ y ∈ s, x ≤ y) : x ≤ s.min' H := H2 _ $ min'_mem _ _
 
+/-- `{a}.min'` is `a`. -/
+@[simp] lemma min'_singleton (a : α) : ({a} : finset α).min' (singleton_nonempty a) = a :=
+by simp [min']
```

Comment:
An alternative here is to add an extra argument `{h}` (no type ascription necessary), and then just write `({a} : finset α).min' h = a`.

I don't think this is anything but a stylistic difference, because of proof irrelevance. To me statements written this way seems slightly easier to parse (because there's no temptation to read the irrelevant proof!)

Reply:
Done, for all three lemmas.